### ISSUE TITLE
missing-optional-annotation requires ? in rest parameter (...foo), but rest parameters cannot be made optional

### ISSUE THREAD

**Issue by fpintos** [Microsoft/tslint-microsoft-contrib#466]
I have a function that looks like this:
```ts
export function foo(p1?: string, ...others:any[]){...}
```

tslint complains with missing-optional-annotation on the "others" parameter:
Argument following optional argument missing optional annotation:  ...others: any[]

However, trying to add ? to it causes the compiler to complain with:
 TS1047: A rest parameter cannot be optional.

This is with 5.10.0.

**Comment by fpintos** [comment]
Same error if I pass an initializer to p1.
My only solution right now is to write the rest parameter code itself inside the function.

**Comment by JoshuaKGoldberg** [comment]
Hey @fpintos - thanks for filing this issue!

The `missing-optional-annotation` rule has been deprecated in this package. The TypeScript compiler will more intelligently keep track of rest parameters. You can disable the rule in your `tslint.json` by deleting this line:

```
"missing-optional-annotation": true,
```

If you haven't yet, I would recommend updating your `tslint-microsoft-contrib` package to at least 5.1.0 to make sure you're not using a version that still recommends that rule.

**Comment by fpintos** [comment]
Cool thanks.
